packages/flutter_tools/lib/src/commands/clean.dart

```diff
@@ -81,11 +86,22 @@ class CleanCommand extends FlutterCommand {
     try {
       final XcodeProjectInterpreter xcodeProjectInterpreter = globals.xcodeProjectInterpreter!;
       final XcodeProjectInfo projectInfo = (await xcodeProjectInterpreter.getInfo(xcodeWorkspace.parent.path))!;
-      for (final String scheme in projectInfo.schemes) {
+      if (argResults?.wasParsed('scheme') ?? false) {
+        final String scheme = argResults!['scheme'] as String;
+        if (scheme.isEmpty) {
+          throwToolExit('No scheme was specified for --scheme');
+        }
+        if (!projectInfo.schemes.contains(scheme)) {
+          throwToolExit('Scheme "$scheme" not found in ${projectInfo.schemes}');
+        }
         await xcodeProjectInterpreter.cleanWorkspace(xcodeWorkspace.path, scheme, verbose: _verbose);
+      } else {
+        for (final String scheme in projectInfo.schemes) {
+          await xcodeProjectInterpreter.cleanWorkspace(xcodeWorkspace.path, scheme, verbose: _verbose);
+        }
       }
     } on Exception catch (error) {
-      globals.printTrace('Could not clean Xcode workspace: $error');
+      throwToolExit('Could not clean Xcode workspace: $error');
```

Comment:
Don't we want the behavior to be the same if the user DIDN'T provide the --scheme option?

Reply:
Ah, good catch. Yes, I think we should only log in the case that `--scheme` wasn't provided. We can throw if it was provided.

Reply:
Agree, I will update that.